### PR TITLE
run: use kwargs instead list of explicit arguments

### PR DESCRIPTION
**Pull request by pared**
* [x] Have you followed the guidelines in our
      [Contributing document](https://dvc.org/doc/user-guide/contributing)?

* [x] Does your PR affect documented changes or does it add new functionality
      that should be documented? If yes, have you created a PR for
      [dvc.org](https://github.com/iterative/dvc.org) documenting it or at
      least opened an issue for it? If so, please add a link to it.

-----
Related to #1900

### CODE REVIEW

File: dvc/repo/run.py [hunk 1]
```diff
@@ -4,69 +4,20 @@
...
 
     if stage is None:
         return None
 
     self.check_dag(self.stages() + [stage])
 
     with self.state:
-        if not no_exec:
-            stage.run(no_commit=no_commit)
+        if not kwargs.get("no_exec", False):
```
Comment (efiop) [inline dvc/repo/run.py]: If we need no_exec, we can simply do `def run(self, no_exec=False, **kwargs)`

**Comment by efiop** [comment]
@pared tests failed

**Comment by Suor** [comment]
What's the purpose? Just to cut on lines?

**Comment by pared** [comment]
@Suor 
1. Cut lines
2. Reduce the number of arguments in methods
3. Don't require passing down arguments while they are used only in one place in code
4. Makes passing new flags/options down the code easier, no need to update every method on the way

File: dvc/stage.py [hunk 2]
```diff
@@ -532,51 +505,52 @@ def create(
             logger.warning("Build cache is ignored when persisting outputs.")
             ignore_build_cache = True
 
-        if validate_state:
```
Comment (efiop) [inline dvc/stage.py]: I guess this one was a leftover from a very old initial pkg implementation and was not used anywhere else, right?

Comment (pared) [reply dvc/stage.py]: yes, I noticed that no use of `create` assigns this value, so its always default

Comment (efiop) [reply dvc/stage.py]: Good catch @pared ! Thanks!

**Pull request merged**